macaroonbakery/macaroon.py

```diff
@@ -226,7 +239,31 @@ def macaroon_version(bakery_version):
     return pymacaroons.MACAROON_V2
 
 
-def parse_local_location(loc):
+class ThirdPartyLocator(object):
+    '''Used to find information on third party discharge services.
+    '''
+    def __init__(self):
+        self._store = {}
+
+    def third_party_info(self, loc):
+        '''Return information on the third party at the given location.
+
+        It returns None if no match is found.
+
+        @param loc string
+        @return: string
+        '''
+        return self._store.get(loc)
+
+    def add_info(self, loc, info):
+        '''Associates the given information with the given location.
+
+        It will ignore any trailing slash.
+        '''
+        self._store[loc.rstrip('\\')] = info
```

Comment:
This strips the wrong kind of slash - we're interested in removing the training / from URLs here.

Reply:
done.